pkg/landscaper/execution/reconcile_test.go

```diff
@@ -167,6 +167,22 @@ var _ = Describe("Reconcile", func() {
 			Expect(err).To(HaveOccurred())
 			Expect(exec.Status.Phase).To(Equal(lsv1alpha1.ExecutionPhaseFailed))
 		})
+
+		It("should set the status of a previously failed execution to progressing if a deployitem has a newer generation than the last observed generation", func() {
```

Comment:
hmm not sure if this should always be the case.
The generation is only updated when something in the spec changes.
But we should also be able to reconcile status changes of the same generation.
Or is this is out-of-scope for this PR?